### PR TITLE
CONTRIBUTING.md: update the description of dependency management

### PR DESCRIPTION
**Pull request by hhxcc**
### What problem does this PR solve? 
update CONTRIBUTING.md  dependency management descrption and go version should be 1.11 or above.

### What is changed and how it works?
update CONTRIBUTING.md  dependency management descrption and go version should be 1.11 or above.

Tests 
 - No code

### CODE REVIEW

**Comment by zz-jason** [comment]
@tiancaiamao PTAL

**Comment by zz-jason** [comment]
@winoros @lamxTyler PTAL

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -175,6 +171,12 @@ git checkout -b myfeature
 
 You can now edit the code on the `myfeature` branch.
 
+#### Download Dependencies
```
Comment (lamxTyler) [inline CONTRIBUTING.md]: Is this needed? If we run the `make server` as in `Run stand-alone mode`, go mod can automatically download it for us.

Comment (hhxcc) [reply CONTRIBUTING.md]: there is no need to do this, I will remove it.

**Pull request merged**